macos/Sources/App/macOS/AppDelegate.swift

```diff
@@ -403,6 +403,10 @@ class AppDelegate: NSObject,
         reloadDockMenu()
     }
 
+    func applicationDidUpdate(_ notification: Notification) {
```

Comment:
Okay I tested the logic of this and imo this callback gets called _way_ too much. On initial launch it's called 4 times and when I close the window once it gets called something like a dozen or more times (I didn't count, but it was a lot). 

I think we need to do something different. Maybe "any open window" isn't the right logic and we need to do some more research into how iTerm2 behaves instead.

Reply:
Yes that's what I noticed as well :/
I saw this method from a blog post about the challenge to track managed windows...
With AppKit Notifications, you get when windows are about to close, and when they become key/main but that's it.

Maybe we could intercept (via swizzling?) when windows are about to be set to the `NSApp.windows` property; or even add KVO to this property (seems that it is not)...

Do you think that `applicationDidUpdate(_:)` may cause performance issues?